crates/pipeline_manager/src/db/mod.rs

```diff
@@ -122,6 +122,31 @@ impl Display for Version {
 #[derive(Debug, Serialize)]
 #[serde(untagged)]
 pub(crate) enum DBError {
+    #[serde(serialize_with = "serialize_pg_error")]
```

Comment:
Let's move everything related to errors to a separate file and module. db/mod.rs is getting crowded.

Reply:
lets do it after we merged all the outstanding PRs though it will be a lot of merge conflict